symengine/sets.cpp

```diff
@@ -620,6 +632,107 @@ RCP<const Set> Complement::set_complement(const RCP<const Set> &o) const
     return container_->set_complement(newuniv);
 }
 
+ConditionSet::ConditionSet(const RCP<const Basic> sym,
+                           RCP<const Boolean> condition)
+    : sym(sym), condition_(condition)
+{
+    SYMENGINE_ASSIGN_TYPEID()
+    SYMENGINE_ASSERT(ConditionSet::is_canonical(sym, condition))
+}
+
+bool ConditionSet::is_canonical(const RCP<const Basic> sym,
```

Comment:
I'm not versed with the theory involved, it should either be `Universal` or `Empty`. @nishnik ?